sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -157,7 +157,14 @@ class NodeHttpClient implements HttpClient {
         if (body && isReadableStream(body)) {
           body.pipe(req);
         } else if (body) {
-          req.end(body);
+          if (typeof body === "string" || Buffer.isBuffer(body)) {
+            req.end(body);
+          } else if (isArrayBuffer(body)) {
+            req.end(ArrayBuffer.isView(body) ? Buffer.from(body.buffer) : Buffer.from(body));
+          } else {
+            logger.error("Unrecognized body type", body);
```

Comment:
This is basically a "never" situation since:

- `FormData`  would be handled by the `isReadableStream` case above
- `Blob` is browser only

There are no other supported body types after that.

Reply:
How is FormData handled by isReadableStream? It doesn't have `pipe` function. also I had thought it's browser only (go to definition opens lib.dom.d.ts) and I didn't find it in nodejs docs.

Reply:
`FormData` (the type you saw) gets converted by this policy when in Node: 

https://github.com/Azure/azure-sdk-for-js/blob/6c59958ebd42c493a96585f8e55aa2257cddafa3/sdk/core/core-rest-pipeline/src/policies/formDataPolicy.ts#L41

It uses an npm package `form-data` to create another object type (confusingly also called `FormData`) that implements the Stream interface.